src/test/java/BaseTest.java

```diff
@@ -1,11 +1,65 @@
 import io.github.bonigarcia.wdm.WebDriverManager;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
 import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.Parameters;
+import java.net.URL;
+import java.time.Duration;
+import java.time.Instant;
 
 public class BaseTest {
+    public WebDriver driver = null;
+    public String url;;
+    WebDriverWait wait;
 
     @BeforeSuite
     static void setupClass() {
         WebDriverManager.chromedriver().setup();
     }
+
+    @BeforeMethod
+    @Parameters({"BaseURL"})
+
+    public void launchBrowser (String BaseURL) {
+        ChromeOptions options = new ChromeOptions();
+        options.addArguments("--remote-allow-origins=*");
+        options.addArguments("--disable-popup-blocking");
+        options.addArguments("--disable-notifications");
+        driver = new ChromeDriver(options);
+        driver.manage().window().maximize();
+        wait = new WebDriverWait(driver,Duration.ofSeconds(10));
+        url = BaseURL;
+    }
+
+    public void navigateToPage() throws InterruptedException {
+        driver.get(url);
+    }
+
+    @AfterMethod
+    public void closeBrowser() throws InterruptedException {
+        driver.quit();
+    }
+    public void provideEmailAddress(String email) throws InterruptedException {
+        WebElement emailField = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("[type='email']")));
+        emailField.clear();
+        emailField.sendKeys(email);
+    }
+
+    public void providePassword(String password) throws InterruptedException {
+        WebElement passwordField = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("[type='password']")));
+        passwordField.clear();
+        passwordField.sendKeys(password);
+    }
+
+    public void clickSubmit() throws InterruptedException {
```

Comment:
Hello,

Your homework 20 has been checked, you're doing great! Keep up the good work!
Best wishes from the HW team (Khanh, khanh@testpro.io)

Test script Passed

![image](https://github.com/MCrowley91/A49-QA-Automation-Framework/assets/44592718/a2635398-6ced-4b52-9dd3-37263f5cd3c9)

But I have suggestion for you: An InterruptedException is thrown when a thread is interrupted while it's waiting, sleeping, or otherwise occupied. In other words, some code has called the interrupt() method on our thread. It's a [checked exception](https://www.baeldung.com/java-checked-unchecked-exceptions), and many blocking operations in Java can throw it.
If you used Explicit waits -> you can remove throws InterruptedException after each function